firewood/src/hashednode.rs

```diff
@@ -310,43 +309,47 @@ impl HasUpdate for Vec<u8> {
     }
 }
 
-pub(super) enum ValueDigest<'a> {
-    /// A node's value.
-    Value(&'a [u8]),
-    /// The hash of a a node's value.
-    /// TODO danlaine: Use this variant when implement ProofNode
-    _Hash(Box<[u8]>),
+#[derive(Clone, Debug)]
+/// A ValueDigest is either a node's value or the hash of its value.
+pub enum ValueDigest<T> {
+    Value(T),
+    /// TODO this variant will be used when we deserialize a proof node
+    /// from a remote Firewood instance. The serialized proof node they
+    /// send us may the hash of the value, not the value itself.
+    _Hash(T),
 }
 
-trait Hashable {
+pub(crate) trait Hashable {
+    type T: AsRef<[u8]>;
```

Comment:
nit: associated types usually have better names, suggest ValueDigestType or something like that

Reply:
Changed to `ValueDigestType`